src/calendarimportmodel.cpp

```diff
@@ -243,7 +227,7 @@ bool CalendarImportModel::importToMemory(const QString &fileName, const QByteArr
             mEventList.append(incidence.staticCast<KCalendarCore::Event>());
     }
     if (!mEventList.isEmpty())
-        qSort(mEventList.begin(), mEventList.end(), incidenceLessThan);
+        std::sort(mEventList.begin(), mEventList.end());
```

Comment:
As far as I know, `KCalCore::Incidence` is not defining any `operator<`. I guess that the `QSharePointer` is defining one, based on memory address. So it is compiling, but it will not do the proper comparison for the sort.